nvfuser/__init__.py

```diff
@@ -54,6 +54,10 @@ def __init__(self, id=None, max_length=1024):
         super(FusionDefinition, self).__init__(id, max_length)
         self.profiled = False
 
+    def segment(self, inputs):
+        self.num_segments = self._setup_segmentation(inputs)
+        self._finalize_segmentation()
```

Comment:
```suggestion
        num_segments = self._setup_segmentation(inputs)
        self._finalize_segmentation()
        return num_segments
```
nitpick: since segment isn't stateful. we probably don't want to keep `num_segments` inside self.

Reply:
`num_segments` is used in the `test_segmentation_reduction_pointwise_epilogue`. We don't have anything else to check in this PR.

Reply:
Ah, I can just check the return value.